### ISSUE TITLE
Can't initialize backend since provider > 2.20

### ISSUE THREAD

**Issue by scalp42** [terraform-providers/terraform-provider-aws#9617]
Not sure if related to https://github.com/terraform-providers/terraform-provider-aws/pull/9428

but trying to initalize backend:

```
AWS_SDK_LOAD_CONFIG=1 AWS_REGION=us-west-2 terraform init                                                                            
Initializing modules...

Initializing the backend...

Error: Error creating AWS session: CredentialRequiresARNError: credential type source_profile requires role_arn, profile bridge
```

It was working fine on Terraform 0.11:

```
$> cat ~/.aws/config                                                                                                                    

[profile dev]
role_arn = arn:aws:iam::xxxxx:role/admin
role_session_name = anthony-dev
source_profile = bridge
region = us-west-2
```

```
$> cat main.tf

provider "aws" {
  alias  = "dev-us-west-2"
  region = "us-west-2"

  profile             = "dev"
  allowed_account_ids = ["xxxxx"]
}
```

Any idea? Thanks in advance!

**Comment by scalp42** [comment]
Just downgraded to `2.20` and it works again.

**Comment by ewbankkit** [comment]
@scalp42 It looks like this is caused by a bug in the AWS SDK - https://github.com/aws/aws-sdk-go/issues/2727 - that has just been merged and should be released in their next version.
The bug was introduced via https://github.com/aws/aws-sdk-go/pull/2694, released in [AWS SDK v1.21.0](https://github.com/aws/aws-sdk-go/releases/tag/v1.21.0), merged into this provider via https://github.com/terraform-providers/terraform-provider-aws/pull/9333 and released in provider [v2.21.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2210-july-25-2019).

**Comment by ewbankkit** [comment]
The fix has just been released in [AWS SDK v1.22.0](https://github.com/aws/aws-sdk-go/releases/tag/v1.22.0).
Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/9555

**Comment by scalp42** [comment]
thanks for sharing @ewbankkit we'll watch for the next release of aws provider, cc @evandam

**Comment by bflad** [comment]
I merged in #9555 earlier today, which should have the upstream AWS Go SDK fix. 🤞 This will release with version 2.23.0 of the Terraform AWS Provider in the next day or two.

Please note that the Terraform S3 Backend code lives upstream in Terraform Core (https://github.com/hashicorp/terraform/tree/master/backend/remote-state/s3) so a separate GitHub issue and SDK bump will be required there.

**Comment by scalp42** [comment]
I opened https://github.com/terraform-providers/terraform-provider-aws/issues/9999 as there are more issues but not sure if they're related.
